TOPIC8-MONGODB/src/main/java/data/CourseDao.java

```diff
@@ -0,0 +1,15 @@
+package data;
+
+import java.util.List;
+
+import models.Course;
+
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.data.mongodb.repository.Query;
+
+public interface CourseDao extends MongoRepository<Course, Long> {
+
+	@Query(value = "{ $and [ { teacherAssigned.idTeacher: ?0 }, { courseName :?0 }}")

```

Comment:
About your comment "I tried to find a course relative to his teacher id and course name , but maybe its wrong", it's ok but you forgot to add the String courseName as a parameter in the method and a correct way to do it would be: public List<Course> findByTeacherIdAndCourseName(Long teacherId, String courseName); if the DB allows you to do that query